CMakeLists.txt

```diff
@@ -260,6 +260,7 @@ add_test(NAME hashtable_test COMMAND hashtable_test)
 add_executable(sumset_test
     tests/sumset_test.c src/sumset.c src/util.c src/trace.c src/hex.c
     src/checksum.c src/rollsum.c src/mdfour.c src/hashtable.c ${blake2_SRCS})
+target_compile_options(sumset_test PRIVATE -Drsync_EXPORTS)
```

Comment:
Ugh! This is because we are statically linking the test, instead of dynamically linking it against the librsync library.

Currently librsync_export.h and librsync.h assumes we only build a librsync shared library and dynamically link against that.

We can't just change sumset_test to dynamically link against the librsync library because it tests functions/methods/data that is not publicly exported by librsync.

So I think we need to change librsync_export.h to support the static linking case. As I understand it, when statically linking the correct thing is to have LIBRSYNC_EXPORT defined as empty.

I'll have a look at best practices and see what I can find.

Reply:
It looks like cmake's GenerateExportHeader (which we can't use because it's broken for C on all but the most recent cmake versions) adds this at the beginning of its librsync_export.h:

```C
#  ifdef LIBRSYNC_STATIC_DEFINE
#    define LIBRSYNC_EXPORT
#    define LIBRSYNC_NO_EXPORT
#  else
```

And then you are expected to define `LIBRSYNC_STATIC_DEFINE` for targets that are statically linked. I think we should copy that... can you include the change to librsync_export.h in this pull request?

Reply:
Added change to `librsync_export.h`, and added `-DLIBRSYNC_STATIC_DEFINE` to `sumset_test` target.